cmake/Reprojucer.cmake

```diff
@@ -6435,6 +6443,9 @@ function(_FRUT_set_cxx_language_standard_properties target)
         else()
           set(cxx_language_standard "20")
         endif()
+      elseif(NOT (DEFINED JUCER_VERSION AND JUCER_VERSION VERSION_LESS 7.0.3)
+          AND JUCER_PROJECT_TYPE STREQUAL "Audio Plug-in")
+        set(cxx_language_standard "17")
```

Comment:
Testing this PR I've hit an issue here. In `jucer_project_settings` I have defined `CXX_LANGUAGE_STANDARD "C++20"` but the code takes this path and my build is being set to C++ 17.

Let me know if you'd like more details or a minimal example to reproduce the issue.

Reply:
Hi @dimitrisudell,
Thanks for catching this! I applied the [changes made to `jucer_ProjectExport_Xcode.h` in JUCE@f8e91d400](https://github.com/juce-framework/JUCE/commit/f8e91d400#diff-f934936ba7b75a0b49560e473a69df8c53b8036933692a68bc47638596ac7462R1707-R1709) without realizing that some of them didn't make a lot of sense. 😅 
Since JUCE 7.0.3 requires C++17 anyway, we don't need these anyway.

Reply:
I've re-tested and this is working as expected now. Thanks for the quick fix.